client/my-sites/checkout/composite-checkout/payment-methods/id-wallet.js

```diff
@@ -0,0 +1,367 @@
+import notices from 'notices';
+import { validatePaymentDetails } from 'lib/checkout/validation';
+import useCountryList from 'my-sites/checkout/composite-checkout/hooks/use-country-list';
+import Field from 'my-sites/checkout/composite-checkout/components/field';
+import {
+	SummaryLine,
+	SummaryDetails,
+} from 'my-sites/checkout/composite-checkout/components/summary-details';
+import { PaymentMethodLogos } from 'my-sites/checkout/composite-checkout/components/payment-method-logos';
+import { maskField } from 'lib/checkout';
+import CountrySpecificPaymentFieldsUI from '../components/country-specific-payment-fields-ui';
+
+const debug = debugFactory( 'composite-checkout:id-wallet-payment-method' );
+
+export function createIdWalletPaymentMethodStore() {
+	debug( 'creating a new id wallet payment method store' );
+	const actions = {
+		changeCustomerName( payload ) {
+			return { type: 'CUSTOMER_NAME_SET', payload };
+		},
+		setFieldValue( key, value ) {
+			return { type: 'FIELD_VALUE_SET', payload: { key, value } };
+		},
+		setFieldError( key, message ) {
+			return { type: 'FIELD_ERROR_SET', payload: { key, message } };
+		},
+		touchAllFields() {
+			return { type: 'TOUCH_ALL_FIELDS' };
+		},
+	};
+
+	const selectors = {
+		getCustomerName( state ) {
+			return state.customerName;
+		},
+		getFields( state ) {
+			return state.fields;
+		},
+	};
+
+	const store = registerStore( 'id-wallet', {
+		reducer(
+			state = {
+				customerName: { value: '', isTouched: false },
+				fields: {},
+			},
+			action
+		) {
+			switch ( action.type ) {
+				case 'CUSTOMER_NAME_SET':
+					return { ...state, customerName: { value: action.payload, isTouched: true } };
+				case 'FIELD_VALUE_SET':
+					return {
+						...state,
+						fields: {
+							...state.fields,
+							[ action.payload.key ]: {
+								value: maskField(
+									action.payload.key,
+									state.fields[ action.payload.key ],
+									action.payload.value
+								),
+								isTouched: true,
+								errors: [],
+							},
+						},
+					};
+				case 'FIELD_ERROR_SET':
+					return {
+						...state,
+						fields: {
+							...state.fields,
+							[ action.payload.key ]: {
+								...state.fields[ action.payload.key ],
+								errors: [ action.payload.message ],
+							},
+						},
+					};
+				case 'TOUCH_ALL_FIELDS':
+					return {
+						...state,
+						fields: Object.keys( state.fields ).reduce(
+							( obj, key ) => ( {
+								...obj,
+								[ key ]: {
+									...state.fields[ key ],
+									isTouched: true,
+								},
+							} ),
+							{}
+						),
+					};
+			}
+			return state;
+		},
+		actions,
+		selectors,
+	} );
+
+	return { ...store, actions, selectors };
+}
+
+export function createIdWalletMethod( { store, stripe, stripeConfiguration } ) {
+	return {
+		id: 'id-wallet',
+		label: <IdWalletLabel />,
+		activeContent: <IdWalletFields stripe={ stripe } stripeConfiguration={ stripeConfiguration } />,
+		submitButton: (
+			<IdWalletPayButton
```

Comment:
Same for `IdWalletPayButton`; it does not need the `stripe` or `stripeConfiguration` data.